src/telemetry/sendloop.cpp

```diff
@@ -36,144 +39,246 @@ void SendLoop::run()
+    writer.Int(battery_data.high_temperature);
+    writer.Key("lowVoltageCell");
+    writer.Int(battery_data.low_voltage_cell);
+    writer.Key("highVoltageCell");
+    writer.Int(battery_data.high_voltage_cell);
+  }
 
+  writer.Key("indvVoltage");
+  writer.StartArray();
   for (int voltage : battery_data.cell_voltage) {
-    battery_data_msg.add_indv_voltage(voltage);
+    writer.Int(voltage);
   }
+  writer.EndArray();
+  writer.EndObject();
 }
 
-void SendLoop::packSensorsMessage(telemetry_data::ClientToServer& msg)
+void SendLoop::packSensorsMessage(Writer<StringBuffer>& writer)
 {
   data::Sensors sensors_data = data_.getSensorsData();
-  telemetry_data::ClientToServer::Sensors* sensors_msg = msg.mutable_sensors();
-
-  sensors_msg->set_module_status(Utils::moduleStatusEnumConversion(sensors_data.module_status));
-
+  writer.Key("sensors");
+  writer.StartObject();
+  writer.Key("moduleStatus");
+  writer.String(convertModuleStatus(sensors_data.module_status).c_str());
+  writer.Key("imu");
+  writer.StartArray();
   for (data::ImuData imu_data : sensors_data.imu.value) {
-    telemetry_data::ClientToServer::Sensors::ImuData* imu_data_msg = sensors_msg->add_imu();
-
-    imu_data_msg->set_operational(imu_data.operational);
+    writer.StartObject();
+    writer.Key("operational");
+    writer.Bool(imu_data.operational);
+    writer.Key("acc");
+    writer.StartArray();
 
     // hardcoded atm to loop three times bc hyped vector doesn't have a method for vector length
     // or have an iterator to support a ranged for loop
     for (int i = 0; i < 3; i++) {
-      imu_data_msg->add_acc(imu_data.acc[i]);
+      writer.Double(imu_data.acc[i]);
     }
+    writer.EndArray();
+    writer.EndObject();
   }
+  writer.EndArray();
+  writer.EndObject();
 }
 
-void SendLoop::packTemperatureMessage(telemetry_data::ClientToServer& msg)
+void SendLoop::packTemperatureMessage(Writer<StringBuffer>& writer)
 {
-  telemetry_data::ClientToServer::Temperature* temp_msg = msg.mutable_temperature();
-
-  temp_msg->set_temperature(data_.getTemperature());
+  writer.Key("temperature");
+  writer.StartObject();
+  writer.Key("temperature");
+  writer.Int(data_.getTemperature());
+  writer.EndObject();
 }
 
-void SendLoop::packEmergencyBrakesMessage(telemetry_data::ClientToServer& msg)
+void SendLoop::packEmergencyBrakesMessage(Writer<StringBuffer>& writer)
 {
   data::EmergencyBrakes emergency_brakes_data = data_.getEmergencyBrakesData();
-  telemetry_data::ClientToServer::EmergencyBrakes* emergency_brakes_msg = msg.mutable_emergency_brakes(); // NOLINT
-
+  writer.Key("emergencyBrakes");
+  writer.StartObject();
+  writer.Key("brakes");
+  writer.StartArray();
   for (bool brake_retracted : emergency_brakes_data.brakes_retracted) {
-    emergency_brakes_msg->add_brakes(brake_retracted);
+    writer.Bool(brake_retracted);
+  }
+  writer.EndArray();
+  writer.EndObject();
+}
+
+std::string SendLoop::convertStateMachineState(data::State state)
```

Comment:
We could put this in a `utils` file maybe later down the road, but not too important

Reply:
Certainly. Also probably going to add some abstraction to the whole data packing, so you could specify the key, data type and value (together with max and min values) or array in one line.